### ISSUE TITLE
Permissions.bin does not work

### ISSUE THREAD

**Issue by dottiekh** [Facepunch/garrysmod-issues#5358]
### Details
Using x86-64 branch.

The permissions.bin file does not update properly at all when being prompted for confirming permissions such as "voicerecord". I'm not sure how limited this is, for I have only had this occur while hosting a local TTT listen server. Once the voicerecord prompt appears and is confirmed, the permissions.bin updates seemingly improperly, populating itself with garbage.

![image](https://user-images.githubusercontent.com/34873184/203035758-9990d096-d2c3-46f6-9772-2bc435a621fb.png)

I'm not sure what exactly the structure of this file is supposed to be, but this doesn't seem like it's properly working.

### Steps to reproduce
1. Activate traitor voice chat while in the TTT gamemode to make the permission prompt appear.
2. Confirm the permission (with "Do not ask me again" checked).
3. Permission does not actually save properly in the file (the game will prompt you again later).

**Comment by robotboy655** [comment]
What OS? Where's your game installed exactly (full path)?

Have you tried non beta branch?

Try deleting your `permissions.bin` file while game is not running and try again and see if it is working or not.

The format is not human readable, so opening it in notepad will not yield good results.

**Comment by dottiekh** [comment]
OS: Win10 21H2
Install dir: F:\Programs\SteamLibrary\steamapps\common\GarrysMod
(F is a secondary drive)

Tested on non-beta branch, permissions are still not being saved properly. The prompt reappears on game restart.

I had figured the file might not be human readable, though it seems like it's not even being populated with anything at all? Forgive me if I'm still misunderstanding what I see in it, but having tested on non beta branch the file only gets filled with a single SOH and NUL character (the second test it was STX and NUL). I imagine that means nothing at all is actually being stored there?

![image](https://user-images.githubusercontent.com/34873184/203154590-ad52179b-313f-438f-985f-2a98ea9a13ba.png)
